lib/rubygems/test_case.rb

```diff
@@ -470,6 +475,13 @@ def teardown
     @back_ui.close
   end
 
+  def credential_setup
+    @temp_cred = File.join(@userhome, '.gem', 'credentials')
+    FileUtils.mkdir_p File.dirname(@temp_cred)
+    File.write @temp_cred, ':rubygems_api_key: 701229f217cdf23b1344c7b4b54ca97'
+    File.chmod 0600, @temp_cred
```

Comment:
Are the tests using `credential_setup` cleaning up after themselves? Ideally we would do this in an `around` hook but minitest doesn't have these so we need a separate `crendential_teardown` method.

Reply:
I did try to use `credential_teardown`. But it caused the many of test failures because the some of tests relied on the credential file implicitly.

Reply:
Do you mind if I have a look and try to fix it?

Reply:
It's super helpful for me. Because you know I'm always busy with the other Ruby works 😇

Reply:
Yes, I'm aware :) I'll look at it later today :+1: